### DISCUSSION TITLE
Using PlanetScope data within FORCE

### DISCUSSION THREAD

**Discussion by dominiqueweber** [davidfrantz/force#299] [Q&A] [answered]
Hi everyone

I would like to use PlanetScope (8-bands) within FORCE for a time series analysis (TSA) (if possible?). As far as I understand, this is basically possible with the force-cube function. I have only found a few examples for Sentinel-1 (here in this forum) and can't get any further with the documentation. Does anyone already have experience with this (or with similar data)? What are the requirements for the input data etc.?

I was also wondering how the cloud mask provided by PLANET could be used for the TSA?

Thanks a lot and best regards
Dominique

**Comment by dominiqueweber** [comment]
Update: The integration of PLANET scenes with force-cube was no problem. However, as far as I understood, further processing, e.g. with the TSA module, is not possible, because PLANET is not supported as a sensor. @davidfrantz Is the integration of PLANET planned or does someone already have a workaround/solution for that?

**Comment by davidfrantz** [comment, marked as answer]
Hi @dominiqueweber,

there is a branch under development that will let you define sensors more freely by yourselves. This will still take quite some time to be finished though.

The currently best solution to bring in Planet data is to disguise them as Sentinel-2. Just rename them as Sentinel-2 with the file naming convention that FORCE uses. When using in TSA, just keep in mind that the requested spectral bands will not match with the actual bands in the data.

- BLUE will use Band 1
- GREEN will use Band 2
- ...
- SWIR1/SWIR2 will crash the program because there are no bands 9 and 10 in your data

The cloud mask is the more tricky part though. It is probably best to apply the cloud filtering to the actual data before going into FORCE (and then use the latest develop version and use ``PRODUCT_TYPE_QUALITY = NULL``, which will disable using quality files). Otherwise, you would need to recreate the QAI bit flag product (although this might be the cleaner approach). 

I know this is not a perfect solution, but I hope it helps somewhat,
David

**Comment by dominiqueweber** [comment]
Thanks for your help @davidfrantz. It works with this workaround. I can now ingest PLANET data and use it for TSA. 


For those who are interested, I have proceeded as follows:
1. Creation of daily cloud-masked composites (multiple PLANET images per day are possible and as far as I understand, FORCE does only work with daily images per sensor)
2. Change filenames following the FORCE naming convention (e.g. 20210703_LEVEL2_SEN2B_BOA.tif) 
3. Re-order bands so that they match the order of Sentinel-2 (at least visible and NIR, for NDVI etc.)
3. Creation of a QAI mask (set everything to "0" that is not NoData due to clouds)
4. force-cube to ingest S2-like PLANET data
5. force-higher-level to run TSA

Cheers, Dominique
